shootout/shootout.py

```diff
@@ -0,0 +1,173 @@
+            return True
+        except ValueError:
+            currency = await bank.get_currency_name(ctx.guild)
+            await ctx.send("Insufficient funds! This game requires {} {}".format(settings["cost"], currency))
+            return False
+
+    async def add_player(self, ctx, cost):
+        current_pot = await self.config.guild(ctx.guild).Session.Pot()
+        await self.config.guild(ctx.guild).Session.Pot.set(value=(current_pot + cost))
+
+        async with self.config.guild(ctx.guild).Session.Players() as players:
+            players.append(ctx.author.id)
+            num_players = len(players)
+
+        if num_players == 1:
+            wait = await self.config.guild(ctx.guild).wait_time()
+            await ctx.send(
+                "{0.author.mention} is gathering players for a shootout!\n"
+                "Type `{0.prefix}shootout` to join in!\n"
+                "The round will start in {1} seconds.".format(ctx, wait)
+            )
+            await asyncio.sleep(wait)
+            await self.start_game(ctx)
+        else:
+            await ctx.send("{} joined the shootout!".format(ctx.author.mention))
+
+    async def start_game(self, ctx):
+        settings = await self.config.guild(ctx.guild).Session.all()
+        players = [ctx.guild.get_member(player) for player in settings["Players"]]
+        filtered_players = [player for player in players if isinstance(player, discord.Member)]
+        if len(filtered_players) < 2:
+            try:
+                await bank.deposit_credits(ctx.author, settings["Pot"])
+            except BalanceTooHigh as e:
+                await bank.set_balance(ctx.author, e.max_balance)
+            await ctx.send("A shootout with yourself means just shooting at some bottles. For that, no charge.")
+            await self.config.guild(ctx.guild).Session.clear()
+            return
+        listen_message = await self.config.guild(ctx.guild).message()
+        await self.config.guild(ctx.guild).Session.Active.set(True)
+        await ctx.send(
+            "The shootout has started! Draw your weapons and type `{}` to shoot your weapon! First person who shoots wins!".format(
+                listen_message
+            )
+        )
+
+        def check(m):
+            return m.content.lower() == listen_message.lower() and m.channel == ctx.channel and m.author in players
+
+        message = await self.bot.wait_for("message", check=check)
+        winner = message.author
+        await self.end_game(ctx, winner)
+
+    async def end_game(self, ctx, winner: discord.Member):
+        currency = await bank.get_currency_name(ctx.guild)
+        total = await self.config.guild(ctx.guild).Session.Pot()
+        try:
+            await bank.deposit_credits(winner, total)
+        except BalanceTooHigh as e:
+            await bank.set_balance(winner, e.max_balance)
+        await ctx.send(
+            "{} has won the shootout! {} {} has been deposited to their account!".format(
```

Comment:
Some randomized win messages would be cool, not a big deal though. Like Russian Roulette